### PR TITLE
feat(csms): add SDK support for CSMS API

### PR DESCRIPTION
**Pull request by chengxiangdong**
Add SDK support for CSMS API.

### CODE REVIEW

File: openstack/csms/v1/secrets/requests.go [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+var RequestOpts = golangsdk.RequestOpts{
+	MoreHeaders: map[string]string{"Content-Type": "application/json", "X-Language": "en-us"},
+}
+
+type CreateSecretOpts struct {
+	Name         string `json:"name" required:"true"`
+	KmsKeyID     string `json:"kms_key_id,omitempty"`
+	Description  string `json:"description,omitempty"`
+	SecretBinary string `json:"secret_binary,omitempty"`
+	SecretString string `json:"secret_string,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/csms/v1/secrets/requests.go:14]: suggest to add `xor` flag:
```
SecretBinary string `json:"secret_binary,omitempty" xor:"SecretString"`
SecretString string `json:"secret_string,omitempty" xor:"SecretBinary"`
```

File: openstack/csms/v1/secrets/requests.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	if err == nil {
+		var r SecretRst
+		rst.ExtractInto(&r)
+		return &r.Secret, nil
+	}
+	return nil, err
+}
+
+type UpdateSecretOpts struct {
+	KmsKeyID    string  `json:"kms_key_id,omitempty"`
```
Comment (ShiChangkuo) [inline openstack/csms/v1/secrets/requests.go:49]: can `kms_key_id` be updated empty which means using the default kms?

File: openstack/csms/v1/secrets/requests.go (hunk 1 above)
Comment (chengxiangdong) [reply openstack/csms/v1/secrets/requests.go:14]: fixed

File: openstack/csms/v1/secrets/requests.go (hunk 2 above)
Comment (chengxiangdong) [reply openstack/csms/v1/secrets/requests.go:49]: If the value is empy, the API returned an error.
If we want to change it, the value must be specified,  even if we change to the default kms.

**Pull request merged**